webpack/components/extensions/RegistrationCommands/fields/ActivationKeys.js

```diff
@@ -39,48 +52,93 @@ const ActivationKeys = ({
 
   // Validate field when hostgroup is changed (host group may have some keys)
   useEffect(() => {
-    handleInvalidField('Activation Keys', akHasValidValue(hostGroupId, pluginValues?.activationKeys, hostGroupActivationKeys));
+    handleInvalidField(
+      'Activation Keys',
+      akHasValidValue(
+        hostGroupId,
+        pluginValues?.activationKeys,
+        hostGroupActivationKeys,
+      ),
+    );
   }, [handleInvalidField, hostGroupId, hostGroupActivationKeys, pluginValues]);
 
+  useEffect(() => {
+    if (activationKeys?.length === 1) {
+      onChange({ activationKeys: [activationKeys[0].name] });
+    }
+  }, [activationKeys, onChange]);
+
   return (
     <FormGroup
+      onFocus={() => setHasInteraction(true)}
       label={__('Activation Keys')}
-      fieldId="activation_keys_field"
-      helperText={hostGroupActivationKeys && sprintf('From host group: %s', hostGroupActivationKeys)}
-      helperTextInvalid={activationKeys?.length === 0 ? <a href="/activation_keys/new">{__('Create new activation key')}</a> : __('No Activation Keys selected')}
-      validated={validateAKField(hostGroupId, pluginValues?.activationKeys, hostGroupActivationKeys)}
-      labelIcon={<LabelIcon text={__('Activation key(s) for Subscription Manager.')} />}
+      fieldId="reg_katello_ak"
+      helperText={
+        hostGroupActivationKeys &&
+        sprintf('From host group: %s', hostGroupActivationKeys)
+      }
+      helperTextInvalid={
+        activationKeys?.length === 0 ? (
+          <a href="/activation_keys/new">{__('Create new activation key')}</a>
+        ) : (
+          __('No Activation Keys selected')
+        )
+      }
+      helperTextInvalidIcon={<ExclamationCircleIcon />}
+      labelIcon={
+        <LabelIcon text={__('Activation key(s) for Subscription Manager.')} />
```

Comment:
Since subscription-manager is going away at some point, this might be a good time to update the wording here.

```suggestion
        <LabelIcon text={__('Activation key(s) to use during registration.')} />
```